src/components/uikit/BottomSheet.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useRef, ReactNode } from 'react';
+import { Dimensions } from 'react-native';
+import RNBottomSheet, {
+  BottomSheetProps as RNBottomSheetProps,
+} from '@gorhom/bottom-sheet';
+
+import { styled, useTheme } from '~styles';
+
+interface BottomSheetProps {
+  initialIndex: number;
+  snapPoints?: string[];
+  children: ReactNode;
+  onSheetChanges?: (index: number) => void;
+  keyboardBehavior?: RNBottomSheetProps['keyboardBehavior'];
+}
+
+export function BottomSheet({
+  initialIndex,
+  snapPoints = ['25%', '50%'],
+  children,
+  onSheetChanges,
+  keyboardBehavior = 'interactive',
+}: BottomSheetProps) {
+  const theme = useTheme();
+  const bottomSheetRef = useRef<RNBottomSheet>(null);
+
+  const handleSheetChanges = useCallback(
+    (index: number) => {
+      if (onSheetChanges) {
+        onSheetChanges(index);
+      }
+    },
+    [onSheetChanges]
+  );
+
+  const { height, width } = Dimensions.get('window');
+
+  return (
+    <Wrapper style={{ height, width }}>
```

Comment:
Is this absolutely positioned wrapper needed? Doesn't the bottom sheet appear on top of everything by default?

Reply:
Nope, surprisingly you can interact with what's behind without it.

Reply:
Could you instead use the backdrop component from the library? https://ui.gorhom.dev/components/bottom-sheet/components/bottomsheetbackdrop